core/src/test/java/com/linecorp/armeria/common/HttpResponseBuilderTest.java

```diff
@@ -0,0 +1,284 @@
+                                             .build();
+        final AggregatedHttpResponse aggregatedRes = res.aggregate().join();
+        assertThat(aggregatedRes.status()).isEqualTo(HttpStatus.OK);
+        assertThat(aggregatedRes.headers().contains("header-1")).isTrue();
+        assertThat(aggregatedRes.headers().contains("header-2")).isTrue();
+        assertThat(aggregatedRes.headers().get("header-1")).isEqualTo("test-value1");
+        assertThat(aggregatedRes.headers().get("header-2")).isEqualTo("test-value2");
+    }
+
+    @Test
+    void buildWithTrailers() {
+        final HttpResponse res = HttpResponse.builder()
+                                             .ok()
+                                             .trailers(HttpHeaders.of("trailer-name",
+                                                                      "trailer-value"))
+                                             .build();
+        final AggregatedHttpResponse aggregatedRes = res.aggregate().join();
+        assertThat(aggregatedRes.status()).isEqualTo(HttpStatus.OK);
+        assertThat(aggregatedRes.trailers().contains("trailer-name")).isTrue();
+        assertThat(aggregatedRes.trailers().get("trailer-name")).isEqualTo("trailer-value");
+    }
+
+    @Test
+    void buildComplex() {
+        final HttpResponse res = HttpResponse.builder()
+                                             .ok()
+                                             .headers(HttpHeaders.of("header-1",
+                                                                     "test-value1",
+                                                                     "header-2",
+                                                                     "test-value2"))
+                                             .header(HttpHeaderNames.ACCEPT_ENCODING, "gzip")
+                                             .header(HttpHeaderNames.ACCEPT_ENCODING, "deflate")
+                                             .header(HttpHeaderNames.ACCEPT_ENCODING, "gzip")
+                                             .content("Armeriaはいろんな使い方がアルメリア")
+                                             .trailers(HttpHeaders.of("trailer-name",
+                                                                      "trailer-value"))
+                                             .build();
+        final AggregatedHttpResponse aggregatedRes = res.aggregate().join();
+        assertThat(aggregatedRes.status()).isEqualTo(HttpStatus.OK);
+        assertThat(aggregatedRes.headers().contains("header-1")).isTrue();
+        assertThat(aggregatedRes.headers().contains("header-2")).isTrue();
+        assertThat(aggregatedRes.headers().get("header-1")).isEqualTo("test-value1");
+        assertThat(aggregatedRes.headers().get("header-2")).isEqualTo("test-value2");
+        assertThat(aggregatedRes.headers().getAll(HttpHeaderNames.ACCEPT_ENCODING))
+                .isEqualTo(ImmutableList.of("gzip", "deflate", "gzip"));
```

Comment:
nit; prefer the assertj api for richer test failure messages (and better style!)

```suggestion
        assertThat(aggregatedRes.headers().getAll(HttpHeaderNames.ACCEPT_ENCODING))
                .containsExactly("gzip", "deflate", "gzip");
```